research/sentiment_analysis/sentiment_main.py

```diff
@@ -0,0 +1,184 @@
+"""The model makes use of concatenation of two CNN layers with different kernel sizes.
+`sentiment_model.py` for more details about the models.
+"""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+# pylint: disable=g-bad-import-order
+from absl import app as absl_app
+from absl import flags
+import tensorflow as tf
+# pylint: enable=g-bad-import-order
+
+from data import dataset
+import sentiment_model
+from official.utils.flags import core as flags_core
+from official.utils.logs import hooks_helper
+from official.utils.logs import logger
+from official.utils.misc import distribution_utils
+
+
+def convert_keras_to_estimator(keras_model, num_gpus, model_dir=None):
+  keras_model.compile(optimizer="rmsprop",
+                      loss="categorical_crossentropy", metrics=["accuracy"])
+
+  distribution = distribution_utils.get_distribution_strategy(
+      num_gpus, all_reduce_alg=None)
+  run_config = tf.estimator.RunConfig(train_distribute=distribution)
+
+  estimator = tf.keras.estimator.model_to_estimator(
+      keras_model=keras_model, model_dir=model_dir, config=run_config)
+
+  return estimator
+
+
+def per_device_batch_size(batch_size, num_gpus):
+  """For multi-gpu, batch-size must be a multiple of the number of GPUs.
+  Note that this should eventually be handled by DistributionStrategies
+  directly. Multi-GPU support is currently experimental, however,
+  so doing the work here until that feature is in place.
+  Args:
+    batch_size: Global batch size to be divided among devices. This should be
+      equal to num_gpus times the single-GPU batch_size for multi-gpu training.
+    num_gpus: How many GPUs are used with DistributionStrategies.
+  Returns:
+    Batch size per device.
+  Raises:
+    ValueError: if batch_size is not divisible by number of devices
+  """
+  if num_gpus <= 1:
+    return batch_size
+
+  remainder = batch_size % num_gpus
+  if remainder:
+    err = ("When running with multiple GPUs, batch size "
+           "must be a multiple of the number of available GPUs. Found {} "
+           "GPUs with a batch size of {}; try --batch_size={} instead."
+           ).format(num_gpus, batch_size, batch_size - remainder)
+    raise ValueError(err)
+  return int(batch_size / num_gpus)
+
+import tensorflow as tf
+import numpy as np
+from data.util import to_dataset, pad_sentence, START_CHAR, OOV_CHAR
+
+
+def run_model():
```

Comment:
I think you are still using FLAGS object directly, and usually the convention is pass the FLAGS as a param here to simplify the unit test by dependency injection. See https://github.com/sculd/models/blob/02590067e6b4e73ef0260114137f47499838ac87/official/resnet/imagenet_main.py#L317